Add building sysext with enki info (#262)
kairos-io/kairos-docs: content/en/docs/Advanced/sys-extensions.md

Some extension examples are available under https://github.com/Itxaka/sysext-examples for k3s and sbctl.

### Building system extensions from a docker image with enki

{{% alert title="Warning" color="warning" %}}
This feature is in preview state and only available in Enki from version 0.1.4
{{% /alert %}}

You can also build a system extension from a docker image directly by using [enki](https://github.com/kairos-io/enki) and using a dockerfile to isolate the artifacts you want converted into a system extension.

Notice that when converting a docker image into a system extension, the last layer is the only one converted (The last command in a given Dockerfile) so have that in mind. This is useful for packages that ONLY install things in /usr or manual installation under /usr

For example for a given Dockerfiles as such:

```dockerfile
FROM anchore/grype:latest AS grype

FROM scratch
COPY --from=grype /grype /usr/local/bin/grype
```

> Only the files added in the last step will be converted to a sysext, so the contents of the sysext would be the `/usr/local/bin/grype` binary only.

Reviewer: Not really. The file with the content `id=_ANY` is auto-generated. We'd better mention that here because while reviewing the PR I didn't know that and it makes is a real pain to create that file too with heredoc in the dockerfile and still keep everything in one command (last layer).

Author: Good point. I did not add it as it seems superfluous to me, but the point that you created the file yourself without knowing that the command would create it automatically says that this info is probably needed here. Will add it somewhere around.

```

The explanation of the docker command flags is as follows:
 - `-v "$PWD"/keys:/keys`: We mount the current dir + keys dir into the container `/keys` path. So Enki has access to the keys to sign the sysext.
 - `-v "$PWD":/build/`: Mount the current dir into the container `/build` path. So the generated sysext is available after the container is gone. 
 - `-v /var/run/docker.sock:/var/run/docker.sock`: We pass the docker sock into the container so it can access our locally built container images. So we avoid pushing them and pulling them from a remote registry.
 - `--rm`: Once the container exit, remove it so we dont leave stuff lying around.

The explanation of the Enki command flags is as follows:
 - `sysext`: Subcommand to call, in this case we want to build a sysext
 - `NAME`: Output and internal name of the sysext.
 - `CONTAINER_IMAGE`: Image from which we will extract the last layer and covert it to a system extension.
 - `--private-key`: Private key to sign the system extension.
 - `--certificate`: Certificate to sign the system extension.

>  - `--output`: Dir where we will output the system extension. Make sure that this matches what is passed to the docker command to be able to keep the generated system extension once the container exists and its removed.

Reviewer: ```suggestion
 - `--output`: Dir where we will output the system extension. Make sure that this matches the directory that passed to the docker command to be able to keep the generated system extension once the container exists and its removed.
```
